kube/src/api/admission.rs

```diff
@@ -147,8 +147,9 @@ pub enum Operation {
 }
 
 /// An outgoing [`AdmissionReview`] response.
-#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq)]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[serde(rename_all = "camelCase")]
+#[non_exhaustive]
 pub struct AdmissionResponse {
```

Comment:
Thoughts on having this as `non-exhaustive` to require using one of the constructors?

Reply:
I think that's reasonable. Then we should also make the patch type private.